engines/wintermute/ad/ad_game.cpp

```diff
@@ -1738,16 +1738,16 @@ Wintermute::TShadowType Wintermute::AdGame::getMaxShadowType(Wintermute::BaseObj
 	return MIN(ret, _scene->_maxShadowType);
 }
 
-bool Wintermute::AdGame::getFogParams(bool *fogEnabled, uint32 *fogColor, float *fogStart, float *fogEnd) {
+bool Wintermute::AdGame::getFogParams(FogParameters &fogParameters) {
 	if (_scene) {
-		*fogEnabled = _scene->_fogEnabled;
-		*fogColor = _scene->_fogColor;
-		*fogStart = _scene->_fogStart;
-		*fogEnd = _scene->_fogEnd;
+		fogParameters._enabled = _scene->_fogEnabled;
```

Comment:
_scene should probably just contain a FogParameters-instance.

Reply:
The FogParameters struct sits in base_renderer_3d.h, which would have to be included in ad_scene.h, inside an #ifdef (since this is ScummVM code). Is this really worth it? Maybe the struct definition could just be moved, but I don't know where.

Reply:
By accident, I had placed the struct definition in base_renderer.h, where it does not belong. Since ad_game.cpp needs the definition, but does not include base_renderer_3d.h, I placed it in dctypes.h now which means that AdScene has it, too.

Reply:
An entirely new header for the purpose wouldn't hurt either.